src/lib.rs

```diff
@@ -208,6 +221,24 @@ impl<T: Send> ThreadLocal<T> {
     /// Returns the element for the current thread, or creates it if it doesn't
     /// exist. If `create` fails, that error is returned and no element is
     /// added.
+    #[cfg(feature = "nightly")]
+    pub fn get_or_try<F, E>(&self, create: F) -> Result<&T, E>
+    where
+        F: FnOnce() -> Result<T, E>,
+    {
+        if let Some(thread) = thread_id::try_get_thread_holder() {
+            if let Some(inner) = self.get_inner(thread.into_inner()) {
+                return Ok(inner);
+            }
+        }
+        let thread = ThreadHolder::new();
```

Comment:
This allocates a new thread ID on *every* insertion, even when one already exists. This is incorrect.

Reply:
I fixed this and also removed the `thread` parameter from the insert call, this improved the `insert` benchmark significantly, but had no impact on the `get` benchmark